### PR TITLE
Trim down the main smoke-testNuGet.Config to generate the tarball version

### PR DESCRIPTION
**Pull request by omajid**
Instead of maintaining the duplicate, just generate it based on the main one.

Fixes: #1669

### CODE REVIEW

File: support/tarball/smoke-testNuGet.Config [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
+﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
     <clear />
-    <add key="source-built-packages" value="SOURCE_BUILT_PACKAGES" />
-    <add key="dotnet-core" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public%40Local/nuget/v3/index.json" />
     <add key="smoke-test feed" value="SMOKE_TEST_PACKAGE_FEED" />
+    <add key="source-built-packages" value="SOURCE_BUILT_PACKAGES" />
+    <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public%40Local/nuget/v3/index.json" />
```
Comment (omajid) [inline support/tarball/smoke-testNuGet.Config:7]: Should I just delete this file from source control? Should I place the generated version at some other location?

Comment (MichaelSimons) [reply support/tarball/smoke-testNuGet.Config:7]: I think it should be deleted.  What purpose would it server?  I think it is going to be a source of confusion?

Comment (MichaelSimons) [reply support/tarball/smoke-testNuGet.Config:7]: Thanks, that just reaffirms my preference to delete it.

Comment (omajid) [reply support/tarball/smoke-testNuGet.Config]: Okay. I have deleted it. The generated one now lives in `artifacts/tarball-smoke-testNuGet.Config`, and in the same location as before in the generated tarball.

**Review by MichaelSimons — APPROVED** [review]

**Review by dseefeld — APPROVED** [review]

**Pull request merged**